transmission_interface/CMakeLists.txt

```
@@ -12,23 +12,24 @@ endif()
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
-find_package(hardware_interface REQUIRED)
-find_package(tinyxml2_vendor REQUIRED)
-find_package(TinyXML2 REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(hardware_interface REQUIRED)    # todo: we should be using this for hardware interface constants
+find_package(tinyxml_vendor REQUIRED)
+find_package(TinyXML REQUIRED)  # provided by tinyxml2 upstream, or tinyxml2_vendor
+
+# todo: we should be using the URDF project to load our tranmissions from URDF
+#find_package(urdf REQUIRED)
 
 add_library(transmission_parser SHARED src/transmission_parser.cpp)
 target_include_directories(transmission_parser PUBLIC include)
-ament_target_dependencies(transmission_parser
-  hardware_interface
-  TinyXML2
-)
+ament_target_dependencies(transmission_parser rclcpp hardware_interface tinyxml_vendor TinyXML)
```

```suggestion
ament_target_dependencies(transmission_parser rclcpp hardware_interface TinyXML2)
```

as [here](https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/CMakeLists.txt)